lua/telescope/pickers.lua

```diff
@@ -446,6 +448,16 @@ function Picker:find()
     end
 
     local prompt = self:_get_prompt()
+
+    local updated_finder
+    if self._on_input_filter_cb then
+      prompt, updated_finder = self._on_input_filter_cb(prompt, self._filter_marker)
+    end
+
+    if updated_finder then
+      self.finder = updated_finder
```

Comment:
I think if we update to a new finder, we should make sure we `close` the old one or something. I can't remember if finders already have a hook for closing them down or not, but we should formalize that if we're going to randomly abandon them at times.

Reply:
I don't see any hook present at the moment. What would this involve?
The finder object consists of the results table and its metatable. Is it enough to set the finder/results tables to nil? Out of curiousity - would this have any practical benefit over abandoning them GC wise?

Reply:
Well some finders start jobs -- they should have the jobs get stopped before we start a new one. Things like that.

We may want to add that before we do this (or I can potentially do it after)

Reply:
Doesn't almost everything other than grep use a `StaticFinder` that doesn't run jobs? It seemed that way when I looked at putting in some job cleanup code.

As it doesn't seem to apply to this context, I'd rather merge sooner rather than later. If the above statement is correct, any issues should only appear in modification of the live-grep(`JobFinder`) finder.  

That way I could leave the jobs cleanup changes for you to do, as you're a lot more familiar with it.